### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by Stratus3D**
[Card](https://rentgroup.atlassian.net/browse/SRV-5981)

This PR was branched off of https://github.com/rentpath/ex_json_logger/pull/24 in order build on what was contributed there.

* Switch to iolist to improvement performance
* Update to latest version of Erlang 26
* Minor performance tweaks
* Document benchee benchmark
* Add Elixir 1.15 and Erlang 26, remove Elixir 1.13 from GitHub workflow

## Benchmarks

<details><summary>Before</summary>

```
Operating System: macOS
CPU Information: Apple M1 Pro
Number of Available Cores: 10
Available memory: 32 GB
Elixir 1.15.7
Erlang 26.2

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 5 s
memory time: 0 ns
reduction time: 0 ns
parallel: 1
inputs: debug, info
Estimated total run time: 14 s

Benchmarking format/4 with input debug ...
Benchmarking format/4 with input info ...

##### With input debug #####
Name               ips        average  deviation         median         99th %
format/4      579.53 K        1.73 μs  ±1081.85%        1.38 μs        2.92 μs

##### With input info #####
Name               ips        average  deviation         median         99th %
format/4      377.40 K        2.65 μs   ±388.45%        2.33 μs        5.08 μs
```
</details>

<details><summary>After</summary>

```
Compiling 1 file (.ex)
Operating System: macOS
CPU Information: Apple M1 Pro
Number of Available Cores: 10
Available memory: 32 GB
Elixir 1.15.7
Erlang 26.2

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 5 s
memory time: 0 ns
reduction time: 0 ns
parallel: 1
inputs: debug, info
Estimated total run time: 14 s

Benchmarking format/4 with input debug ...
Benchmarking format/4 with input info ...

##### With input debug #####
Name               ips        average  deviation         median         99th %
format/4        1.55 M      643.44 ns  ±2151.32%         583 ns         750 ns

##### With input info #####
Name               ips        average  deviation         median         99th %
format/4      991.21 K        1.01 μs  ±1398.08%        0.92 μs        1.13 μs
```
</details>

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -42,7 +42,8 @@ defmodule ExJsonLogger.Mixfile do
       {:credo, "~> 1.6.7", only: [:dev, :test]},
       {:dialyxir, "~> 0.5.1", only: [:dev], runtime: false},
       {:plug, "~> 1.14"},
-      {:jason, "~> 1.0"}
+      {:jason, "~> 1.0"},
+      {:benchee, "~> 1.2", only: :bench}
```
Comment (pasha) [inline mix.exs:46]: i think we should only load this in dev

**Review by rkledesma — APPROVED** [review]

**Review by rkledesma — COMMENTED** [review]
Some failing unit tests

**Comment by Stratus3D** [comment]
> Some failing unit tests

I fixed the tests locally. CI workflow is running tests on all versions of Erlang.

File: .github/workflows/elixir.yml [hunk 2]
```diff
@@ -8,11 +8,11 @@ jobs:
...
       matrix:
-        otp: ['23', '24', '25']
-        elixir: ['1.13', '1.14']
+        otp: ['24', '25', '26']
+        elixir: ['1.14', '1.15']
         include:
-          - otp: '22'
-            elixir: '1.13'
+          - otp: '23'
+            elixir: '1.14'
```
Comment (Stratus3D) [inline .github/workflows/elixir.yml:15]: I needed to add Erlang 26 and Elixir 1.15 to this matrix. I didn't see any reason to continue supporting Elixir 1.13.

File: mix.exs (hunk 1 above)
Comment (Stratus3D) [reply mix.exs]: I tested the benchmarks with environment changed to `dev` and got the same numbers so I guess there is no harm in changing it to `dev`. Just note that in `dev` benchee will always be downloaded when running `mix deps.get`. PR updated.

**Review by pasha — APPROVED** [review]

**Pull request merged**